framework/decode/vulkan_replay_consumer_base.cpp

```
@@ -2865,6 +2865,34 @@ VulkanReplayConsumerBase::OverrideCreateInstance(VkResult original_result,
     VkResult result =
         create_instance_proc_(&create_state.modified_create_info, GetAllocationCallbacks(pAllocator), replay_instance);
 
+    if (result == VK_ERROR_EXTENSION_NOT_PRESENT)
+    {
+        // Assume we weren't able to create instance because of VK_KHR_portability_enumeration
+        GFXRECON_LOG_WARNING(
+            "Assuming unable to create instance due to VK_KHR_portability_enumeration. Attempting creation "
+            "without that extension...");
+        create_state.modified_create_info.flags &= ~VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
+
+        std::vector<const char*>& modified_extensions = create_state.modified_extensions;
+        for (int i = 0; i < modified_extensions.size(); ++i)
+        {
+            const char* name = modified_extensions[i];
+            if (!util::platform::StringCompare(name, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME))
+            {
+                const size_t last_idx         = modified_extensions.size() - 1;
+                modified_extensions[i]        = modified_extensions[last_idx];
+                modified_extensions[last_idx] = name;
+                break;
+            }
+        }
+        create_state.modified_extensions.pop_back();
+        create_state.modified_create_info.enabledExtensionCount -= 1;
+
+        // Try to create instance again
+        result = create_instance_proc_(
```

maybe here it makes sense to check if we actually removed `VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME`? (check if `enabledExtensionCount` has changed since last attempt)

---

What should be done if there was no change? Just log an error and let it crash on the next attempt?

---

just logging an error seems appropriate. I mean we know we added the portability-extension, so 'expect' it gets removed here. the log-error 'could' remind us of that assumption if that changes one day.

---

this case basically does not exist. we always start with `VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME` in the extension list. 
I was thinking if that 'could' change one day, we shouldn't get confused here. 
so, not too relevant, was not an actual concern.